pandas/core/ops.py

```diff
@@ -567,6 +567,84 @@ def mask_cmp_op(x, y, op, allowed_types):
 # Functions that add arithmetic methods to objects, given arithmetic factory
 # methods
 
+def _get_flex_wrappers(cls):
+    """
+    Find the appropriate operation-wrappers to use when defining flex
+    arithmetic or boolean operations with the given class.
+
+    Parameters
+    ----------
+    cls : class
+
+    Returns
+    -------
+    arith_method : function or None
+    comp_method : function or None
+
+    Notes
+    -----
+    None is only returned for SparseArray
+    """
+    if issubclass(cls, ABCSeries):
+        # Same for Series and SparseSeries
+        # Note this check must come before check for ABCSparseArray
+        # or else if will incorrectly catch SparseSeries.
+        arith_method = _flex_method_SERIES
+        comp_method = _flex_method_SERIES
+    elif issubclass(cls, ABCSparseArray):
+        arith_method = None
+        comp_method = None
+    elif issubclass(cls, ABCPanel):
+        arith_method = _flex_method_PANEL
+        comp_method = _comp_method_PANEL
+    elif issubclass(cls, ABCDataFrame):
+        # Same for DataFrame and SparseDataFrame
+        arith_method = _arith_method_FRAME
+        comp_method = _flex_comp_method_FRAME
+    return arith_method, comp_method
+
+
+def _get_special_wrappers(cls):
+    """
+    Find the appropriate operation-wrappers to use when defining special
+    arithmetic, comparison, or boolean operations with the given class.
+
+    Parameters
+    ----------
+    cls : class
+
+    Returns
+    -------
+    arith_method : function
+    comp_method : function
+    bool_method : function
```

Comment:
I think it would be better just have a dict here keyed on the cls and just returns the 5 methods. This seems a bit overkill

Reply:
Sounds good.

Reply:
Implementing this it ends up being pretty close to a wash.  We have to key on `cls._typ` most of the time but also check for `cls._subtyp` for SparseSeries and SparseDataFrame.  At this point I'm leaning towards "being a little bit verbose here is OK in this case".  The main upside is its helpful to see which wrappers are used where without having to jump around the file.

Reply:
ok, can do in a single function then